src/Coordination/KeeperStorage.h

```diff
@@ -8,188 +8,376 @@
 
 #include <absl/container/flat_hash_set.h>
 
+#include "config.h"
+#if USE_ROCKSDB
+#include <Coordination/RocksDBContainer.h>
+#endif
+
 namespace DB
 {
 
 class KeeperContext;
 using KeeperContextPtr = std::shared_ptr<KeeperContext>;
 
-struct KeeperStorageRequestProcessor;
-using KeeperStorageRequestProcessorPtr = std::shared_ptr<KeeperStorageRequestProcessor>;
+/// struct KeeperStorageRequestProcessor;
+/// using KeeperStorageRequestProcessorPtr = std::shared_ptr<KeeperStorageRequestProcessor>;
 using ResponseCallback = std::function<void(const Coordination::ZooKeeperResponsePtr &)>;
 using ChildrenSet = absl::flat_hash_set<StringRef, StringRefHash>;
 using SessionAndTimeout = std::unordered_map<int64_t, int64_t>;
 
-struct KeeperStorageSnapshot;
-
-/// Keeper state machine almost equal to the ZooKeeper's state machine.
-/// Implements all logic of operations, data changes, sessions allocation.
-/// In-memory and not thread safe.
-class KeeperStorage
+struct KeeperRocksNodeInfo
```

Comment:
Some basic info are the same, but rocksdb does not need some fields like `children`, and need some auxiliary field like `serialized`. that is the difference.

Besides, we have to seperate `data` and `KeeperRocksNodeInfo`. because `KeeperRocksNodeInfo` is POD so that we can serialize it directly.